### ISSUE TITLE
Create a pre-submit builder to test .ci.yaml roller

### ISSUE THREAD

**Issue by keyonghan** [flutter/flutter#101390]
The .ci.yaml roller could fail due to invalid syntax from landed PRs, and this error gets only caught in post-submit roller builder. This will cause costly roll back or forward fix by blocking the tree.

We should have a pre-submit builder to validate .ci.yaml changes and the roller logic.

AI from go/xcode-update-postmortem

**Comment by CaseyHillers** [comment]
The issue is the engine has the roller marked as `bringup: true` as it pulls from ToT flutter and cocoon. Some context in https://github.com/flutter/engine/pull/28134

We need to update the recipe to use an AOT version of generate_jspb so it only needs to make a request to CIPD instead of checking out these repos.

I made generate_jspb buildable, but depriotiized the work as having generic builders would remove the need to have the roller in presubmit. I uploaded a version in https://chrome-infra-packages.appspot.com/p/flutter/ci_yaml/generate_jspb/linux-amd64/+/ but it isn't being generated automatically on Cocoon's post-submit.
